sui/suiptb/programmable_transaction_builder.go

```diff
@@ -77,8 +77,11 @@ func (p *ProgrammableTransactionBuilder) Obj(objArg ObjectArg) (Argument, error)
 		}
 
 		switch {
-		case oldObjArg.SharedObject.InitialSharedVersion == objArg.SharedObject.InitialSharedVersion:
-			if oldObjArg.id() != objArg.id() {
+		case oldObjArg.SharedObject != nil && objArg.SharedObject != nil &&
+			oldObjArg.SharedObject.InitialSharedVersion == objArg.SharedObject.InitialSharedVersion:
+			oldId := oldObjArg.id()
```

Comment:
I think I prefer to add `Equals()` method to `ObjectId`, and call the method here.
Are you willing to do the change? Or I can merge this PR and add the following change

Reply:
make sense.

Reply:
Btw @cfal I have been thinking about restructure the project. Not sure whether the current naming of each package is straightforward enough? And maybe some features are placed not in the most proper package right now. 
If you have any idea, please let me know

Reply:
Thank you a lot